static/app/components/deviceName.tsx

```diff
@@ -1,81 +1,81 @@
 import * as React from 'react';
+import * as Sentry from '@sentry/react';
 
+// Self reference to the module, so that we can mock a failed import in a test.
+import * as selfModule from 'sentry/components/deviceName';
 import {IOSDeviceList} from 'sentry/types/iOSDeviceList';
 
-export function deviceNameMapper(model: string, iOSDeviceList): string {
-  const modelIdentifier = model.split(' ')[0];
-  const modelId = model.split(' ').splice(1).join(' ');
-  const modelName = iOSDeviceList.generationByIdentifier(modelIdentifier);
-  return modelName === undefined ? model : modelName + ' ' + modelId;
+export function deviceNameMapper(
+  model: string | undefined,
+  module: IOSDeviceList | null
+): string | null {
+  // If we have no model, render nothing
+  if (typeof model !== 'string') {
+    return null;
+  }
+
+  // If module has not loaded yet, render the unparsed model
+  if (module === null) {
+    return model;
+  }
+
+  const [identifier, ...rest] = model.split(' ');
+
+  const modelName = module.generationByIdentifier(identifier);
+  return modelName === undefined ? model : `${modelName} ${rest.join(' ')}`;
 }
 
-export async function loadDeviceListModule() {
+export async function loadDeviceListModule(platform: 'iOS') {
```

Comment:
since the two places using this function only pass `iOS`, should we make it a default parameter?

Reply:
I think it's ok not to. I wouldn't want someone to run `loadDeviceListModule()` in the future and wonder why their Android name resolutions are not working 😅